grails-app/controllers/org/pih/warehouse/api/StockMovementApiController.groovy

```diff
@@ -827,9 +827,10 @@ class StockMovementApiController {
 
     // Function for getting appropriate filter options based on current list and supporting requests approval
     List<RequisitionStatus> getOutboundRequisitionStatusCodes(Boolean isApprovalRequired, Boolean isElectronicType) {
-        // If a location doesn't have approval required, return listOutboundOptions no matter what list is displayed
+        // If a location doesn't have approval required, return listOutboundOptions when we are on outbound list
+        // but if we are on the requests list return listRequestsOptions
         if (!isApprovalRequired) {
-            return RequisitionStatus.listOutboundOptions()
+            return isElectronicType ? RequisitionStatus.listRequestOptions() : RequisitionStatus.listOutboundOptions()
```

Comment:
I think a large part of the problem is that Approval should have been implemented as a separate state machine. This is becoming clear to me because we should be able to apply the approval model to any of our activities: requests, purchase orders, supplier preferences, receipts, etc.